include/boost/beast/core/impl/flat_buffer.hpp

```diff
@@ -262,17 +260,32 @@ reserve(std::size_t n)
 template<class Allocator>
 void
 basic_flat_buffer<Allocator>::
-shrink_to_fit()
+shrink_to_fit() noexcept
 {
     auto const len = size();
+
     if(len == capacity())
         return;
+
     char* p;
     if(len > 0)
     {
         BOOST_ASSERT(begin_);
         BOOST_ASSERT(in_);
-        p = alloc(len);
+#ifndef BOOST_NO_EXCEPTIONS
+        try
+        {
+#endif
+            p = alloc(len);
+#ifndef BOOST_NO_EXCEPTIONS
+        }
+        catch(std::exception const&)
+        {
+            // swallow this exception
```

Comment:
The optics on swallowing followed by an abortion are not great

Reply:
:-D